fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/rollcall/RollCallCreationFragment.java

```diff
@@ -89,54 +90,23 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
     setupConfirmButton();
     setupOpenButton();
     setupCancelButton();
-
-    // Subscribe to "new LAO event creation" event
-    mLaoDetailViewModel
-        .getNewLaoEventCreationEvent()
-        .observe(
-            getViewLifecycleOwner(),
-            eventTypeEvent -> {
-              EventType eventType = eventTypeEvent.getContentIfNotHandled();
-              if (eventType == EventType.ROLL_CALL) {
-                createRollCall(false);
-              }
-            });
-
-    // Subscribe to "open new roll call" event
-    mLaoDetailViewModel
-        .getOpenNewRollCallEvent()
-        .observe(
-            getViewLifecycleOwner(),
-            booleanEvent -> {
-              Boolean action = booleanEvent.getContentIfNotHandled();
-              if (action != null) {
-                createRollCall(true);
-              }
-            });
-
-    mLaoDetailViewModel
-        .getCreatedRollCallEvent()
-        .observe(
-            getViewLifecycleOwner(),
-            booleanEvent -> {
-              Boolean action = booleanEvent.getContentIfNotHandled();
-              if (action != null) {
-                mLaoDetailViewModel.openLaoDetail();
-              }
-            });
   }
 
   private void setupConfirmButton() {
-    confirmButton.setOnClickListener(
-        v -> mLaoDetailViewModel.newLaoEventCreation(EventType.ROLL_CALL));
+    confirmButton.setOnClickListener(v -> createRollCall(false));
   }
 
   private void setupOpenButton() {
-    openButton.setOnClickListener(v -> mLaoDetailViewModel.openNewRollCall(true));
+    openButton.setOnClickListener(v -> createRollCall(true));
   }
 
   private void setupCancelButton() {
-    mFragBinding.rollCallCancel.setOnClickListener(v -> mLaoDetailViewModel.openLaoDetail());
+    mFragBinding.rollCallCancel.setOnClickListener(
+        v ->
+            setCurrentFragment(
```

Comment:
maybe we need a setParentFragment() where we pass the Fragment manager as argument, and a setActivityFragment() where the default getSupportFragmentManager() is called in the implementation ?